MCprep_addon/util.py

```diff
@@ -48,31 +48,45 @@
 # GENERAL SUPPORTING FUNCTIONS (no registration required)
 # -----------------------------------------------------------------------------
 
-
-def apply_colorspace(node: Node, color_enum: Tuple) -> None:
-	"""Apply color space in a cross compatible way, for version and language.
-
-	Use enum nomeclature matching Blender 2.8x Default, not 2.7 or other lang
+def apply_noncolor_data(node: Node) -> Optional[MCprepError]:
+	"""
+	Apply the Non-Color/Generic Data option to the passed
+	node in a way that is cross version compatible, as well as OCIO
+	config compatible in theory.
+	
+	Returns:
+		If success: None
+		If fail: 
+			All cases - MCprepError(TypeError)
 	"""
-	global noncolor_override
-	noncolor_override = None
+	options: List[str] = []
+	if env.json_data:
+		options = env.json_data["non_color_options"]
 
 	if not node.image:
 		env.log("Node has no image applied yet, cannot change colorspace")
-
-	# For later 2.8, fix images color space user
-	if hasattr(node, "color_space"):  # 2.7 and earlier 2.8 versions
-		node.color_space = 'NONE'  # for better interpretation of specmaps
-	elif hasattr(node.image, "colorspace_settings"):  # later 2.8 versions
-		# try default 'Non-color', fall back to best guess 'Non-Colour Data'
-		if color_enum == 'Non-color' and noncolor_override is not None:
-			node.image.colorspace_settings.name = noncolor_override
-		else:
+	
+	# While earlier versions of 2.8 used a different 
+	# attribute name for colorspace, MCprep officially 
+	# supports the full 2.8 release at a minimum (as of 3.5)
```

Comment:
I think it's enough to just say this should work for 2.80+, don't need to keep comments around for what things "used" to do (that's what changelogs are for)